### PR TITLE
fix(checks): correct session-settings, quiet idle-ratio, make replication-lag capacity-aware

### PR DESCRIPTION
**Pull request by emancu**
Reduces false positives and fixes correctness across three checks. **All queries remain PostgreSQL 14+ compatible** (verified against a live `postgres:14`; `transaction_timeout` / `max_slot_wal_keep_size` handled gracefully across versions).

## session-settings (correctness)
- **Precedence**: encode the full `pg_db_role_setting` precedence — `role+db > role > db > ALTER ROLE ALL > reset_val` — instead of only catching cluster-wide role overrides. The reported value now matches what a role actually gets on connect. (Note: role-wide overrides database-wide, per the PG docs.)
- **Unit parsing**: `ALTER ROLE ... SET statement_timeout='2000ms'` stores the value literally, so the old `strconv.ParseInt` crashed and skipped the whole check. New unit-aware parser handles `2000ms`, `2s`, `1min`, `1.5s`, bare numbers, and `-1`/`0` sentinels → milliseconds.
- **Version gating**: `transaction_timeout` is PG17+. On PG<17 it is absent from `pg_settings`, so the check now **skips** it instead of reporting a false `MUST be set` FAIL for every role.

## connection-health / idle-ratio (noise)
- Now **advisory**: warn at `>=90%` idle, **no FAIL tier**. A high idle ratio is a harmless oversized-pool hint; genuine connection exhaustion is already covered by `connection-saturation` and `pool-pressure`.

## replication-lag — logical (noise + capacity awareness)
- `replay_lag` time tracks Debezium's LSN-ack cadence (batching), not danger. WARN/FAIL now require **both** sustained time **and** a material backlog: `>=120s + >=550 MiB` (warn), `>=300s + >=2 GiB` (fail).
- New **capacity-relative** signal: compares the backlog against `max_slot_wal_keep_size` (`>=50%` warn, `>=85%` fail), firing **before** Postgres flips `wal_status` to `unreserved`. Disabled when the cap is `-1` (unlimited / RDS default).
- Severity = the max of the two tiers. Physical replication unchanged.

## Notes
- `db/query.sql.go` is regenerated; this also drops the orphan `MissingProviderIdTables` (no source query/check exists for it).
- Tests cover the parser edge cases, the version-gated skip, the idle-ratio downgrade, and the logical-lag tiers (incl. real production samples that should stay OK).

### CODE REVIEW

File: checks/sessionsettings/check.go [hunk 1]
```diff
@@ -370,19 +375,80 @@ func (s dbSessionSettings) hasRole(role string) bool {
...
+	"h":   3600000,
+	"d":   86400000,
+}
+
+// parseDurationMs converts a pg_settings value to integer milliseconds.
+//
+// ALTER ROLE stores values literally, so value may carry a unit suffix
+// ("2000ms", "1.5s", "2000 ms") or be a bare number interpreted in baseUnit.
+// The sign is preserved so sentinel values like -1 (disabled) survive.
+func parseDurationMs(value, baseUnit string) (int64, error) {
```
Comment (emancu) [inline checks/sessionsettings/check.go]: 🤔 dont we have a function to do this already? In the whole library?

File: checks/sessionsettings/export_test.go
Comment (emancu) [inline checks/sessionsettings/export_test.go:1]: 🤔 What is this file, i dont think we need it. I think the function we defined internally already exists (or similar) somewhere else

File: checks/sessionsettings/check.go (hunk 1 above)
Comment (emancu) [reply checks/sessionsettings/check.go]: Good question — I checked the whole library and there's **no existing parser**. What exists in `check/format.go` is `FormatDurationMs` (ms → "1.5h"), the *formatter* — the inverse direction. Every other check reads `pg_settings.setting`, which Postgres already normalises to a bare base-unit integer (no suffix), so they just `ParseInt` it (e.g. `vacuumsettings.fetchInt64`). session-settings is the odd one out because it reads **`pg_db_role_setting` override values** — raw `ALTER ROLE ... SET` literals that can carry a unit suffix (`'2000ms'`, `'2s'`). Postgres also has no SQL builtin for this (`pg_size_bytes` is memory-only, not time), so it can't be pushed into the query.

But it *does* have a natural home: I moved it next to its inverse as exported **`check.ParseDurationMs`** (ba6658b), so it's reusable and lives where `FormatDurationMs` does. Behaviour unchanged.

File: checks/sessionsettings/export_test.go
Comment (emancu) [reply checks/sessionsettings/export_test.go:1]: You're right, we don't. It only existed to expose the unexported `parseDurationMs` to the external `_test` package. Now that the parser is exported as `check.ParseDurationMs`, the shim is unnecessary — **deleted in ba6658b**, with its unit tests moved to a new `check/format_test.go` (which also adds the first test coverage for `format.go`).
